### ISSUE TITLE
[BUG] Compatibility check fails for USER-INTEL kspace when using restart files and hybrid pair styles.

### ISSUE THREAD

**Issue by donatas-surblys** [lammps/lammps#1927]
**Summary**

Using the USER-INTEL kspace requires a compatible pair style. Currently, LAMMPS fails with a "ERROR: Intel styles for kspace require intel pair style.", when using pair_style hybrid and read_restart, even though no such errors are present when using a data file or making a system from scratch.

**LAMMPS Version and Platform**

Latest LAMMPS version (33996d9bac98d4bf2b852531a93ebe495b584d7f). Compiled with intel version 18.0.2 20180210 using Makefile.intel_cpu_intelmpi.

**Steps to Reproduce**

First, create a simple restart (and data) file for a system with hybrid pair with the -sf intel argument. Note that no errors are produced.
```
units real
atom_style charge

region box block -30 30 -30 30 -30 30

create_box 1 box

pair_style hybrid/overlay lj/cut/coul/long 14 10 morse 10
kspace_style pppm 1e-6

mass 1 100
create_atoms 1 single 0 0 -1
create_atoms 1 single 0 0  1
set atom 1 charge  1
set atom 2 charge -1

pair_coeff 1 1 lj/cut/coul/long 1 1
pair_coeff 1 1 morse 1 1 1

run 0

write_data data.sys
write_restart restart.sys
```

LAMMPS errors out when trying to use the previously constructed restart file when executing with the -sf intel argument.
```
units real
atom_style charge
kspace_style pppm 1e-6

# this doesn't work
read_restart restart.sys
pair_coeff 1 1 lj/cut/coul/long 1 1
pair_coeff 1 1 morse 1 1 1
run 0

# this works
#pair_style hybrid/overlay lj/cut/coul/long 14 10 morse 10
#read_data data.sys
#pair_coeff 1 1 lj/cut/coul/long 1 1
#pair_coeff 1 1 morse 1 1 1
#run 0

# this also works
#read_restart restart.sys
#pair_style hybrid/overlay lj/cut/coul/long 14 10 morse 10
#pair_coeff 1 1 lj/cut/coul/long 1 1
#pair_coeff 1 1 morse 1 1 1
#run 0
```
The error can be avoided by using a data file or redefining pair styles after read_restart.

**Expected Behavior**

LAMMPS should correctly detect the compatible pair style when using read_restart.

**Actual Behavior**

Currently, LAMMPS produces the following error:

> ERROR: Intel styles for kspace require intel pair style. (../fix_intel.cpp:551)

**Further Information, Files, and Links**

This issue is related to #1808, which was fixed in PR #1817. I think that read_restart is calling Force::new_pair, which is trying and failing to find a hybrid/overlay/intel style.

**Comment by akohlmey** [comment]
> This issue is related to #1808, which was fixed in PR #1817. I think that read_restart is calling Force::new_pair, which is trying and failing to find a hybrid/overlay/intel style.

There is no "hybrid/overlay/intel" pair style. As far as I can tell, the issue is in pair style hybrid itself, and thus in hybrid/overlay as well. Sub-styles for hybrid must be stored without suffix when using the suffix flag, so that pair_coeff can work, as it uses the (non-suffix) pair style name to map individual coeff settings to sub-styles. Consequently, pair style hybrid must apply the suffix when reading the restart, which it was not doing. Defining the pair style before reading the restart will work, as the pair style names match and thus the style is not replaced but only the settings renewed. The following 1-line change should fix this issue:
```
  diff --git a/src/pair_hybrid.cpp b/src/pair_hybrid.cpp
  index fe319da9e..4d8d6f690 100644
  --- a/src/pair_hybrid.cpp
  +++ b/src/pair_hybrid.cpp
  @@ -742,7 +742,7 @@ void PairHybrid::read_restart(FILE *fp)
       keywords[m] = new char[n];
       if (me == 0) utils::sfread(FLERR,keywords[m],sizeof(char),n,fp,NULL,error);
       MPI_Bcast(keywords[m],n,MPI_CHAR,0,world);
  -    styles[m] = force->new_pair(keywords[m],0,dummy);
  +    styles[m] = force->new_pair(keywords[m],1,dummy);
       styles[m]->read_restart_settings(fp);
       // read back per style special settings, if present
       special_lj[m] = special_coul[m] = NULL;
```

**Comment by donatas-surblys** [comment]
Yes, the patch fixes my errors. Thank you.
